src/metrics/include/metrics/metrics.hpp

```diff
@@ -292,14 +296,24 @@ class RegistryT final {
         if (phase) {
             str << '.' << *phase;
         }
-        return str.str();
+
+        const auto& metricName = str.str();
+        if (_ignoredMetrics.count(metricName) == 1) {
+            str << ".__METRICS_IGNORED__";
```

Comment:
1. But this doesn't actually cause us to ignore, it just appends a sentinel?
2. Would it make sense to lean into the existing `canary_` prefix logic instead  and treat ignored metrics as canaries? I don't know the full downstream implications of overloading canaries logic like that, but there might be rails there somewhere this could ride.
3. The user could do this today without this change by simply specifying `MetricsName: foo.__IGNORED__`. In essence this is yaml syntax sugar. As such, if we want to pursue this "ignore via suffix" strategy, I suggest we consider putting this logic into the existing [preprocessing python](https://github.com/mongodb/genny/blob/4e6e1c7da4d830f44fbf4a65c4d4e3bae2011a6a/src/lamplib/src/genny/tasks/preprocess.py). We've found syntax translations like this much easier to reason about, write, and test in Python and thus keep the C++-side of things kinda "dumb."